forch/faucetizer.py

```diff
@@ -131,31 +135,38 @@ def _faucetize(self):
             for port, port_map in switch_map.get('interfaces', {}).items():
                 if 'stack' in port_map or 'lacp' in port_map or 'output_only' in port_map:
                     continue
-                port_map['native_vlan'] = self._config.unauthenticated_vlan
-
-        devices = {**self._dynamic_devices, **self._static_devices}
-        for mac, device in devices.items():
-            if device.placement.switch and device.behavior.segment:
-                switch_cfg = behavioral_faucet_config.get('dps', {}).get(
-                    device.placement.switch, {})
-                port_cfg = switch_cfg.get('interfaces', {}).get(device.placement.port)
-
-                if not port_cfg:
-                    LOGGER.warning('Switch or port not defined in faucet config: %s, %s',
-                                   device.placement.switch, device.placement.port)
-                    continue
 
-                vid = self._segments_to_vlans.get(device.behavior.segment)
-                if not vid:
-                    LOGGER.warning('Device segment does not have a matching vlan: %s, %s',
-                                   mac, device.behavior.segment)
-                    continue
+                port_map['native_vlan'] = self._config.unauthenticated_vlan
```

Comment:
Can you extract the logic above for if it's an acceptable port above to an appropriately named function?  Would like to know what it's calculating.  Also, not sure that logic is sound e.g. not everything will use lacp.  How about also checking for tagged_vlans?

Reply:
Moved to class method '_is_access_port' and added checking for 'tagged_vlans'.